src/main/java/org/sagebionetworks/web/client/security/AuthenticationControllerImpl.java

```diff
@@ -213,56 +234,56 @@ public UserProfile getCurrentUserProfile() {
 	}
 
 	@Override
-	public String getCurrentUserSessionToken() {
-		return currentUserSessionToken;
+	public String getCurrentUserAccessToken() {
+		return currentUserAccessToken;
 	}
 
 	@Override
-	public void signTermsOfUse(boolean accepted, AsyncCallback<Void> callback) {
-		userAccountService.signTermsOfUse(getCurrentUserSessionToken(), accepted, callback);
+	public void signTermsOfUse(AsyncCallback<Void> callback) {
+		userAccountService.signTermsOfUse(getCurrentUserAccessToken(), callback);
 	}
 
 	@Override
 	public void checkForUserChange() {
-		String oldUserSessionToken = currentUserSessionToken;
-		initializeFromExistingSessionCookie(new AsyncCallback<UserProfile>() {
+		checkForUserChange(null);
+	}
+	
+	@Override
+	public void checkForUserChange(Callback optionalCallback) {

```

Comment:
Added an optional callback, which is called after the access token has been initialized in the app.  This is useful for app initialization (in Portal.java).  If not set, then it has the same behavior as before (which the SessionDetector calls periodically).

Reply:
Could we rename callback to webAppInitCallback to indicate it's only used for init?